include/oneapi/dpl/pstl/hetero/dpcpp/execution_sycl_defs.h

```diff
@@ -311,6 +286,36 @@ using __enable_if_device_execution_policy_double_no_default =
                            __is_convertible_to_event<_Events...>,
                        _T>;
 
+template <typename _BackendTag>
+struct __hetero_tag
+{
+    using __backend_tag = _BackendTag;
+};
+
+struct __device_backend_tag
+{
+};
+
+template <class... _IteratorTypes, typename _KernelName>
+::std::enable_if_t<__is_random_access_iterator_v<_IteratorTypes...>, __hetero_tag<__device_backend_tag>>
+__select_backend(const execution::device_policy<_KernelName>&, _IteratorTypes&&...)
+{
+    return {};
+}
+
+#if _ONEDPL_FPGA_DEVICE
+struct __fpga_backend_tag : __device_backend_tag
+{
+};
+
+template <class... _IteratorTypes, unsigned int _Factor, typename _KernelName>
+::std::enable_if_t<__is_random_access_iterator_v<_IteratorTypes...>, __hetero_tag<__fpga_backend_tag>>
```

Comment:
Does `__is_random_access_iterator_v` do `std::decay_t` underneath?

Reply:
Practically yes, because it's based on `__is_iterator_of` :
```C++
template <typename _IteratorTag, typename... _IteratorTypes>
auto
__is_iterator_of(int) -> decltype(
    ::std::conjunction<::std::is_base_of<
        _IteratorTag, typename ::std::iterator_traits<::std::decay_t<_IteratorTypes>>::iterator_category>...>{});
```